homeassistant/components/tradfri/sensor.py

```diff
@@ -67,3 +81,46 @@ def __init__(
     def _refresh(self) -> None:
         """Refresh the device."""
         self._attr_native_value = self.coordinator.data.device_info.battery_level
+
+
+class AirQualitySensor(TradfriBaseEntity, SensorEntity):
+    """
+    The platform class required by Home Assistant.
+
+    Following the discussion here, the sensor seem to be reporting in microgram per
+    m3:
+    https://github.com/dresden-elektronik/deconz-rest-plugin/
+    issues/5351#issuecomment-968362604
+    """
+
+    _attr_device_class = SensorDeviceClass.AQI
+    _attr_native_unit_of_measurement = CONCENTRATION_MICROGRAMS_PER_CUBIC_METER
+
+    def __init__(
+        self,
+        device_coordinator: TradfriDeviceDataUpdateCoordinator,
+        api: Callable[[Command | list[Command]], Any],
+        gateway_id: str,
+    ) -> None:
+        """Initialize a air quality sensor."""
+        super().__init__(
+            device_coordinator=device_coordinator,
+            api=api,
+            gateway_id=gateway_id,
+        )
+
+        self._refresh()  # Set initial state
+
+        self._attr_unique_id = f"{self._attr_unique_id}-air_quality"
+
+    def _refresh(self) -> None:
+        """Refresh the device."""
+        aqi_value = self.coordinator.data.air_purifier_control.air_purifiers[
```

Comment:
Let's refactor and use entity descriptions instead. We can define a lambda or function and set it as a custom entity attribute, and use this attribute to get the correct value from the coordinator.

Example:
https://github.com/home-assistant/core/blob/696b930b1c11d934af047bde3351f6ac17bf1208/homeassistant/components/airly/sensor.py#L115

Then we can use the same entity class for all sensors. We CSN create a map between the device control attribute and the description key and look up what entities with what description to create.

Reply:
Oh, so we want a generic sensor instead of two different types?

Reply:
I’ve think I’ve seen those descriptions elsewhere, I’ll investigate.

Reply:
I just got my intellifire integration approved - we went through a similar process

Reply:
> Let's refactor and use entity descriptions instead. We can define a lambda or function and set it as a custom entity description attribute, and use this attribute to get the correct value from the coordinator.
> 
> Example:
> 
> https://github.com/home-assistant/core/blob/696b930b1c11d934af047bde3351f6ac17bf1208/homeassistant/components/airly/sensor.py#L115
> 
> Then we can use the same entity class for all sensors. We can create a map between the device control attribute and the description key and look up what entities with what description to create.

Not quite sure I followd you on the mapping but I've implemented basic functionality for entity descriptions.